src/services/data/handlers/sink.ts

```diff
@@ -108,175 +109,120 @@ export const seatool: Handler = async (event) => {
-          submitterName: null,
-        };
+  // is Legacy
+  if (record?.origin !== "micro") {
+    if (record?.sk !== "Package") return null;
+    if (!record.submitterName) return null;
+    if (record.submitterName === "-- --") return null;
 
-        oneMacRecords.push(oneMacTombstone);
+    const result = transformOnemacLegacy(id).safeParse(record);
+    return result.success ? result.data : null;
+  }
 
-        console.log(
-          `Record ${id} has been nullified with the following data: `,
-          JSON.stringify(oneMacTombstone)
-        );
-      }
-    }
+  // is new create
+  if (!record?.actionType) {
+    const result = transformOnemac(id).safeParse(record);
+    return result.success ? result.data : null;
   }
+
+  // --------- Package-Actions ---------//
+
+  //ENABLE_RAI_WITHDRAW
+  if (record.actionType === Action.ENABLE_RAI_WITHDRAW) {
+    const result = toggleWithdrawRaiEnabledSchema.safeParse(record);
+    return result.success ? result.data : null;
+  }
+  //DISABLE_RAI_WITHDRAW
+  if (record.actionType === Action.DISABLE_RAI_WITHDRAW) {
+    const result = toggleWithdrawRaiEnabledSchema.safeParse(record);
+    return result.success ? result.data : null;
+  }
+  //ISSUE_RAI
+  if (record.actionType === Action.ISSUE_RAI) {
+    const result = transformRaiIssue(id).safeParse(record);
+    return result.success ? result.data : null;
+  }
+  //RESPOND_TO_RAI
+  if (record.actionType === Action.RESPOND_TO_RAI) {
+    const result = transformRaiResponse(id).safeParse(record);
+    return result.success ? result.data : null;
+  }
+  //WITHDRAW_RAI
+  if (record.actionType === Action.WITHDRAW_RAI) {
+    const result = transformRaiWithdraw(id).safeParse(record);
+    return result.success ? result.data : null;
+  }
+  //WITHDRAW_PACKAGE
+  if (record.actionType === Action.WITHDRAW_PACKAGE) {
+    const result = transformWithdrawPackage(id).safeParse(record);
+    return result.success ? result.data : null;
+  }
+
+  return null;
+};
+
+export const onemac_main = async (event: Event) => {
+  const records = Object.values(event.records).reduce((ACC, RECORDS) => {
+    RECORDS.forEach((REC) => {
+      const dataTransform = onemacDataTransform(REC);
+      if (!dataTransform) return;
+      ACC.push(dataTransform);
+    });
+
+    return ACC;
+  }, [] as any[]);
+
   try {
-    await os.bulkUpdateData(osDomain, "main", oneMacRecords);
+    await os.bulkUpdateData(osDomain, "main", records);
   } catch (error) {
     console.error(error);
   }
 };
+
+export const onemac_changelog = async (event: Event) => {
+  const data = Object.values(event.records).reduce((ACC, RECORDS) => {
+    RECORDS.forEach((REC) => {
+      //TODO: handle delete
+      if (!REC.value) return;
+
+      // NO transform??
+      const packageId = decode(REC.key);
+      const timestamp = new Date().toISOString();
+      ACC.push({
+        ...JSON.parse(decode(REC.value)),
+        id: `${packageId}-${timestamp}`,
+        packageId,
+      });
+    });
```

Comment:
@mdial89f 

The format of id is open to discussion.  I have it here interpolated with the timestamp

Reply:
Updated - replaced timestamp w/ offset